src/main/java/org/folio/gobi/HelperUtils.java

```diff
@@ -90,54 +91,74 @@ public static String encodeValue(String value) throws UnsupportedEncodingExcepti
         Field field = Field.valueOf(mapping.getField().toString());
         org.folio.rest.mappings.model.DataSource ds = mapping.getDataSource();
-        String combinator = ds.getCombinator();
-        NodeCombinator nc = null;
-        if (combinator != null) {
+        DataSource dataSource = extractOrderMapping(ds, map);
+        map.put(field, dataSource);
+      }
+    }
+
+    return map;
+  }
+
+  public static DataSource extractOrderMapping(org.folio.rest.mappings.model.DataSource dataSource, Map<Field, DataSource> map){
+    org.folio.rest.mappings.model.DataSource.Combinator combinator = dataSource.getCombinator();
+    NodeCombinator nc = null;
+    if (combinator != null) {
+      try {
+        Method combinatorMethod = Mapper.class.getMethod(combinator.toString(), NodeList.class);
+        nc = data -> {
           try {
-            Method combinatorMethod = Mapper.class.getMethod(combinator, NodeList.class);
-            nc = data -> {
-              try {
-                return (String) combinatorMethod.invoke(null, data);
-              } catch (Exception e) {
-                logger.error("Unable to invoke combinator method: " + combinator, e);
-              }
-              return null;
-            };
-          } catch (NoSuchMethodException e) {
-            logger.error("Combinator method not found: " + combinator, e);
+            return (String) combinatorMethod.invoke(null, data);
+          } catch (Exception e) {
+            logger.error("Unable to invoke combinator method: " + combinator, e);
           }
-        }
-        String translation = ds.getTranslation();
-        Translation<?> t = null;
-        if (translation != null) {
+          return null;
+        };
+      } catch (NoSuchMethodException e) {
+        logger.error("Combinator method not found: " + combinator, e);
+      }
+    }
+    org.folio.rest.mappings.model.DataSource.Translation translation = dataSource.getTranslation();
+    Translation<?> t = null;
+    if (translation != null) {
+      try {
+
+        Method translationMethod = Mapper.class.getMethod(translation.toString(), String.class);
+        t = data -> {
           try {
-            Method translationMethod = Mapper.class.getMethod(translation, String.class);
-            t = data -> {
-              try {
-                return (CompletableFuture<Object>) translationMethod.invoke(null, data);
-              } catch (Exception e) {
-                logger.error("Unable to invoke translation method: " + translation, e);
-              }
-              return null;
-            };
-          } catch (NoSuchMethodException e) {
-            logger.error("Translation method not found: " + translation, e);
+            return (CompletableFuture<Object>) translationMethod.invoke(null, data);
+          } catch (Exception e) {
+            logger.error("Unable to invoke translation method: " + translation, e);
```

Comment:
it might be helpful to log additional context... e.g. the full mapping or at least the field